kernel/kernel-api/src/main/java/io/delta/kernel/internal/replay/LogReplay.java

```diff
@@ -388,50 +387,66 @@ private Map<String, DomainMetadata> loadDomainMetadataMap(Engine engine) {
   }
 
   /**
-   * Calculates the latest snapshot hint before or at the current snapshot version, returns the
+   * Calculates the latest snapshot hint before or at the current snapshot version, cache the
    * CRCInfo if checksum file at the current version is read
+   *
+   * <p>Strategy for finding and using CRC (checksum) files:
+   *
+   * <ol>
+   *   <li>During initial log directory listing, the SnapshotManager captures all files including
+   *       commits, checkpoints, and checksums.
+   *   <li>The most recent checksum file discovered is stored in the LogSegment as
+   *       "lastSeenChecksum", regardless of its version, however, logSegment will make sure
```

Comment:
can you delete the `regardless of its version` -- since that's not true -- the 2nd half of this sentence is good and accurate